### PR TITLE
[5.6] Adds database count assertion

### PR DESCRIPTION
**Pull request by dallincoons**
This adds a count assertion helper to `InteractsWithDatabase`

### CODE REVIEW

**Comment by sisve** [comment]
I at least miss the $message parameter for a custom message, but I would also like a $delta parameter to add some fuzziness to the comparison.

File: src/Illuminate/Foundation/Testing/Constraints/HasDatabaseCount.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    /**
+     * Check if the data is found in the given table.
+     *
+     * @param  int  $count
+     * @return bool
+     */
+    public function matches($count)
+    {
+        return $this->database->table($this->table)->where($this->data)->count() == $count;
```
Comment (carusogabriel) [inline src/Illuminate/Foundation/Testing/Constraints/HasDatabaseCount.php]: Please use `===`, let’s be strict 🛡

File: src/Illuminate/Foundation/Testing/Constraints/HasDatabaseCount.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    /**
+     * Get the description of the failure.
+     *
+     * @param  int  $count
+     * @return string
+     */
+    public function failureDescription($count)
+    {
+        return sprintf(
+            'count of %d rows in table [%s] matches the attributes %s',
```
Comment (carusogabriel) [inline src/Illuminate/Foundation/Testing/Constraints/HasDatabaseCount.php]: This message could be `count of x row in table y is z`

**Review by carusogabriel — CHANGES_REQUESTED** [review]

File: src/Illuminate/Foundation/Testing/Constraints/HasDatabaseCount.php (hunk 2 above)
Comment (deleugpn) [reply src/Illuminate/Foundation/Testing/Constraints/HasDatabaseCount.php]: `Expected [$count] rows in table [$table], found [$x]`

Comment (mfn) [reply src/Illuminate/Foundation/Testing/Constraints/HasDatabaseCount.php]: 👎 

Why? I bet `[$table]` is only done because the existing constraints do it, but they don't "special quote" (?) other dynamic values.

I wouldn't even know why especially the table needs to be in `[``]` but since existing code does it, consistency is good; doesn't mean this has to be carried forward especially there's no prior code doing that.

TL;DR: my sentiments would be to leave the PR as it is, it's most consistent with existing code 👍

**Comment by taylorotwell** [comment]
I rather just people run database queries and run assertEquals on the expected count.
